ambry-mysql/src/main/resources/AmbryContainerStorageStats.ddl

```diff
@@ -19,43 +19,70 @@
  */
 CREATE TABLE IF NOT EXISTS AccountReports
 (
-    clusterName  VARCHAR(25) NOT NULL,
-    hostname     VARCHAR(30) NOT NULL,
-    partitionId  INT         NOT NULL,
-    accountId    INT         NOT NULL,
-    containerId  INT         NOT NULL,
-    storageUsage BIGINT      NOT NULL,
-    updatedAt    TIMESTAMP   NOT NULL,
+    clusterName          VARCHAR(25) NOT NULL,
+    hostname             VARCHAR(30) NOT NULL,
+    partitionId          INT         NOT NULL,
+    accountId            INT         NOT NULL,
+    containerId          INT         NOT NULL,
+    storageUsage         BIGINT      NOT NULL, /* logical storage usage */
+    physicalStorageUsage BIGINT      NOT NULL, /* physical storage usage */
+    numberOfBlobs        BIGINT      NOT NULL,
+    updatedAt            TIMESTAMP   NOT NULL,
 
     PRIMARY KEY (clusterName, hostname, partitionId, accountId, containerId),
     INDEX updatedAtIndex (updatedAt)
 )
     CHARACTER SET utf8
     COLLATE utf8_bin;
 
+/**
+  We have to add physicalStorageUsage and numberOfBlobs to the existing AccountReports table.
+  We would like to set physicalStorageUsage's value to be the same as storageUsage and the numberOfBlobs to be 0
+
+  ALTER TABLE AccountReports ADD numberOfBlobs BIGINT NOT NULL DEFAULT 0;
+  ALTER TABLE AccountReports ADD physicalStorageUsage BIGINT NOT NULL DEFAULT 0;
+  Update AccountReports SET physicalStorageUsage = storageUsage;
+*/
+
 /**
  * This table is created to store aggregated account storage usage. An aggregation task will be activated to read all
  * the container storage usages from table AcountReports and generate an aggregated stats and write it back to this table.
  */
 CREATE TABLE IF NOT EXISTS AggregatedAccountReports
 (
-    clusterName  VARCHAR(25) NOT NULL,
-    accountId    INT         NOT NULL,
-    containerId  INT         NOT NULL,
-    storageUsage BIGINT      NOT NULL,
-    updatedAt    TIMESTAMP   NOT NULL,
+    clusterName          VARCHAR(25) NOT NULL,
+    accountId            INT         NOT NULL,
+    containerId          INT         NOT NULL,
+    storageUsage         BIGINT      NOT NULL, /* logical storage usage */
+    physicalStorageUsage BIGINT      NOT NULL, /* physical storage usage */
+    numberOfBlobs        BIGINT      NOT NULL,
```

Comment:
Will this be the number of valid blobs or the total number of blobs in storage, including those awaiting compaction?

Reply:
It includes blobs at all states, including those to be compacted.